text/0000-slice-copy-fill.md

```diff
@@ -0,0 +1,99 @@
+- Feature Name: std::slice::copy, slice::fill
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Safe `memcpy` from one slice to another of the same type and length, and a safe
+`memset` of a slice of type `T: Copy`.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the only way to quickly copy from one non-`u8` slice to another is to
+use a loop, or unsafe methods like `std::ptr::copy_nonoverlapping`. This allows
+us to guarantee a `memcpy` for `Copy` types, and is safe. The only way to
+`memset` a slice, currently, is a loop, and we should expose a method to allow
+people to do this. This also completely gets rid of the point of
+`std::slice::bytes`, which means we can remove this deprecated and useless
+module.
+
+# Detailed design
+[design]: #detailed-design
+
+Add two methods to Primitive Type `slice`.
+
+```rust
+impl<T> [T] where T: Copy {
+    pub fn fill(&mut self, value: T);
+    pub fn copy_from(&mut self, src: &[T]);
+}
+```
+
+`fill` loops through slice, setting each member to value. This will lower to a
+memset in all possible cases. It is defined behavior to call `fill` on a slice
+which has uninitialized members, and `self` is guaranteed to be fully filled
+afterwards.
+
+`copy` panics if `src.len() != self.len()`, then `memcpy`s the members into 
```

Comment:
"`self` is guaranteed to be fully filled afterwards". This is important for unsafe code. It is easy to slice down `src` in client code if necessary.